### DISCUSSION TITLE
Prisma connect many to many is not working

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3568] [Q&A] [answered]
Any idea why this connection is not working?

Model:
```prisma
model Promotion {
  id              String                @default(cuid()) @id
  subscriptions   RenamedSubscription[] @relation("RelationPromotionSubscription")
}


model RenamedSubscription {
  id                      String                   @default(cuid()) @id
  promotions              Promotion[]              @relation("RelationPromotionSubscription")
  @@map("Subscription")
}

```

Code to do the connection:
```ts
    let promotion = await ctx.prisma.promotion.update({
      where: {
        id: promotion.id,
      },
      data: {
        subscriptions: {
          connect: {
            id: subscription.id,
          },
        },
      },
      include: {
        subscriptions: true,
      },
    })
    console.log(promotion)
```

the output is:
```js
promotion
{
  id: 'ckcmn2uwtngsx0803z27bvqss',
  subscriptions: [] // <--- It is empty here! It should not!
}
```

No errors have been dropped. `subscription.id` and `promotion.id` exist.

**Comment by ryands17** [comment, marked as answer]
Hey @alan345 👋 

I tried reproducing this with the above model, but it's working for me. The following is my query:

```ts
let promotion = await prisma.promotion.create({
    data: {},
  })
  let subscription = await prisma.renamedSubscription.create({
    data: {},
  })

  let updatedPromotion = await prisma.promotion.update({
    where: {
      id: promotion.id,
    },
    data: {
      subscriptions: {
        connect: {
          id: subscription.id,
        },
      },
    },
    include: {
      subscriptions: true,
    },
  })
  console.log(JSON.stringify(updatedPromotion, null, 2))
```

And here's the output:
![image](https://user-images.githubusercontent.com/19697099/92698695-59638a00-f36a-11ea-97f2-29d562f91f17.png)

I'm on version _2.6.2_ of `@prisma/cli` and `@prisma/client`. Could you verify again from your end?

**Reply by alan345** [reply]
I am also 2.6.2 of `@prisma/cli` and `@prisma/client`.


The only difference with your code is my table is called `Subscription`. Which is a reserved word. So I used: `@@map("Subscription")` inside the schema `RenamedSubscription`. It might be the issue..

Also in your schema, I have a relationship name: `@relation("RelationPromotionSubscription")`. Do you have it? it says here: (https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#example-2) `An implicit m-n-relation does not require @relation attribute.`

And finally can show me the structure of your SQL Join table between `Promotion` and `Subscription`?
Mine is

```sql
CREATE TABLE `_RelationPromotionSubscription` (
  `id` char(25) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  `A` char(25) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  `B` char(25) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `id_UNIQUE` (`id`),
  UNIQUE KEY `AB_unique` (`A`,`B`),
  KEY `A` (`A`),
  KEY `B` (`B`),
  CONSTRAINT `_RelationPromotionSubscription_ibfk_1` FOREIGN KEY (`A`) REFERENCES `Promotion` (`id`) ON DELETE CASCADE,
  CONSTRAINT `_RelationPromotionSubscription_ibfk_2` FOREIGN KEY (`B`) REFERENCES `Subscription` (`id`) ON DELETE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
```

**Reply by ryands17** [reply]
Your join table is incorrect. You need to create it as described [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations/#conventions-for-relation-tables-in-implicit-m-n-relations).

So my entire setup looks like:

```sql
create table "Promotion" (
	id serial primary key,
	name text
);

create table "Subscription" (
	id serial primary key,
	name text
);

create table "_PromotionToSubscription" (
    "A" integer not null references "Promotion" (id),
    "B" integer not null references "Subscription" (id)
);

create unique index "_PromotionToSubscription_AB_unique" on "_PromotionToSubscription" ("A", "B");

create index "_PromotionToSubscription_B_index" on "_PromotionToSubscription" ("B");
```

Then run `prisma introspect` as usual.

This is how you would get implicit relations and your query will now work :)

**Reply by omar-dulaimi** [reply]
@ryands17 @janpio I'm having the same issue at version 2.18.0

No error message, no warnings, nothing.

As you know our project is upgraded from Prisma 1. So the structure for the implicit m-m wasn't touched during the upgrade. I added more details on Slack here: https://prisma.slack.com/archives/CKQTGR6T0/p1618147859354400

**Reply by janpio** [reply]
Best open a new discussion or issue about this @omar-dulaimi (still link to this one for context) - then someone can take a new look at what is happening for you.

**Reply by omar-dulaimi** [reply]
Thanks for the tip. We solved the issue over Slack with Ryan.

**Reply by da201501245** [reply]
We are facing the same problem. how exactly you guys have solved this problem?  currently, we have `id` column as the primary key in the m-m relational table in the prisma1 table.

**Comment by alan345** [comment]
Great ! it works. It could have been nice to get an error dropped when Prisma doesn't manage to write data because the database tables are not "compliant"

**Reply by janpio** [reply]
We are happy to tackle that. Can you create a new issue with a minimal reproduction of the problem you encountered? Then we can investigate that and see why you did not get the helpful error message.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
